packages/astro/src/transitions/router.ts

```diff
@@ -9,12 +9,17 @@ type State = {
 };
 type Events = 'astro:page-load' | 'astro:after-swap';
 
+// Create bound versions of pushState/replaceState so that Partytown doesn't hijack them,
+// which breaks Firefox.
+const pushState = history.pushState.bind(history);
+const replaceState = history.replaceState.bind(history);
+
```

Comment:
Looks good to me. I have no idea, how party town scripts are injected. But these assignments should be early enough as `router.ts` is loaded through `ViewTransitions.astro` from the `<head>`.

Reply:
Should be guarded with `if (inBrowser)...`

Reply:
Does the router ever run outside of a browser?

Reply:
not run but loaded in SSR, see #8771

Reply:
.. and in that failing test :)